src/app/storage/duck/sagas.ts

```diff
@@ -0,0 +1,265 @@
+import { takeLatest, select, race, take, call, put, delay, } from 'redux-saga/effects';
+import { ClientFactory } from '../../../client/client_factory';
+import { IClusterClient } from '../../../client/client';
+import { MigResource, MigResourceKind } from '../../../client/resources';
+import {
+  CoreNamespacedResource,
+  CoreNamespacedResourceKind,
+} from '../../../client/resources';
+import {
+  createStorageSecret,
+  createMigStorage,
+  updateMigStorage,
+  updateStorageSecret,
+} from '../../../client/resources/conversions';
+import { Creators } from './actions';
+import { alertErrorTimeout } from '../../common/duck/actions';
+import {
+  createAddEditStatus,
+  AddEditState,
+  AddEditMode,
+  IAddEditStatus,
+  AddEditWatchTimeout,
+  AddEditWatchTimeoutPollInterval,
+  AddEditConditionCritical,
+  createAddEditStatusWithMeta,
+  AddEditConditionReady,
+  AddEditDebounceWait,
+} from '../../common/add_edit_state';
+
+function* addStorageRequest(action)  {
+  // TODO: Need to improve this to fall into the failed create state with rollback
```

Comment:
if I fill in the form with bogus values, the add storage button does not disable or give the user any visual feedback 

![image](https://user-images.githubusercontent.com/11218376/62080861-70efba00-b21f-11e9-95be-afdac9e5fd33.png)

Reply:
![image](https://user-images.githubusercontent.com/11218376/62081331-5c5ff180-b220-11e9-8ce1-cfcc8499866e.png)

storage add is throwing 400 but appears to be connected after refreshing the page